rfd/0131-adminitrative-actions-mfa.md

```diff
@@ -0,0 +1,274 @@
+---
+authors: Brian Joerger (bjoerger@goteleport.com)
+state: draft
+---
+
+# RFD 131 - Administrative Actions MFA
+
+## Required Approvers
+
+- Engineering: @r0mant && @codingllama
+- Product: @klizhentas || @xinding33
+- Security: @jentfoo
+
+## What
+
+Enforce MFA verification for any administrative actions taken. For example:
+
+- Adding, removing, or editing Nodes and other services.
+- Adding, removing, or editing users, roles, auth connectors, and other
+administrative resources.
+- Approving access requests.
+
+This will apply to administrative actions performed via the WebUI, `tctl`,
+and `tsh`, or Teleport Connect.
+
+## Why
+
+Currently, Teleport only prompts for MFA verification in a few circumstances:
+
+1. User logs in with MFA.
+2. User updates their password or registered MFA devices.
+3. User connects to a service that requires Per-session MFA.
+
+As a result, the vast majority of Teleport actions do not require timely MFA
+verification. If an admin's login session is compromised, it can be used to
+perform attacks pertaining to privilege escalation, security downgrades, Auth
+connector changes, and other forms of cluster sabotage.
+
+Adding an MFA restriction to administrative actions limits this attack vector
+by re-verifying the user's identity promptly before performing any administrative
+action that could be abused.
+
+## Details
+
+### Server configuration
```

Comment:
I think we need to hide this behind some feature flag and have users explicitly enroll rather than making default behavior for all existing installs. It will be a breaking change otherwise.

Reply:
The reason I made this automatically enabled with MFA is that this seems to be the intention from [the issue](https://github.com/gravitational/teleport/issues/9089):

```
In 8.0 if MFA is enrolled, Teleport UI asks for MFA authentication for any administrative action.
```

It seems like this is something we wanted with MFA initially but it slipped until now.

> Cluster settings are a strong indicator of the presence of MFA devices, so barring direct auth access this could be safe (or safe "enough") to enable without causing problems.

Actually, the enforcement will not be based in the cluster settings, rather the presence of a registered MFA device for the user itself, so it should never be applied to a user who cannot complete an MFA challenge.